stac_fastapi/core/stac_fastapi/core/core.py

```diff
@@ -507,13 +507,26 @@ async def all_catalogs(
                 break
             token = next_token
 
+        if catalog_path:
+            catalog_url = f"catalogs/{catalog_path}"
+        else:
+            catalog_url = ""
+
         links = [
-            {"rel": Relations.root.value, "type": MimeTypes.json, "href": base_url},
-            {"rel": Relations.parent.value, "type": MimeTypes.json, "href": base_url},
+            {
+                "rel": Relations.root.value,
+                "type": MimeTypes.json,
+                "href": urljoin(base_url, catalog_url),
```

Comment:
Shouldn't the root link still be `base_url`?

Reply:
I thought this should be the base of the current catalogue?
The STAC client can then use the parent links to jump between catalogues, instead.

Reply:
The root link should be to the top Catalog, ie /api/catalogue/stac/. A number of things in the STAC standards would seem a bit strange if it's interpreted another way.

Reply:
Understood, updated to use `baseurl`